bob-access/src/error.rs

```diff
@@ -0,0 +1,98 @@
+use axum::{
+    response::{IntoResponse, Response},
+    Json,
+};
+use http::StatusCode;
+use serde_json::json;
+use std::{error::Error as StdError, fmt::Display};
+use tonic::{Code, Status};
+
+#[derive(Debug)]
+pub enum Error {
+    _Unknown,
+    InvalidToken(String),
+    Validation(String),
+    Os(String),
+    UserNotFound,
+    ConversionError(String),
+    CredentialsNotProvided(String),
+    MultipleCredentialsTypes,
+    UnauthorizedRequest,
+    PermissionDenied,
+}
+
+impl Error {
+    pub fn msg(&self) -> &'static str {
+        use Error::*;
+        match self {
+            _Unknown => "Unknown error",
+            InvalidToken(_) => "Invalid token",
+            Validation(_) => "Validation error",
+            Os(_) => "Os error",
+            UserNotFound => "User not found",
```

Comment:
It is not safe to return for client this message, because it can be used to find out which users are registered and which are not. It is better to return `Unauthorized request` message here (same as for `UnauthorizedRequest`)